addons/web/static/tests/views/fields/many2one_field_tests.js

```diff
@@ -3238,6 +3238,31 @@ QUnit.module("Fields", (hooks) => {
         );
     });
 
+    QUnit.test("no_quick_create option on a many2one when can_create is absent", async function (assert) {
+        serverData.models.partner.fields.product_id.readonly = true;
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <form>
+                    <sheet>
+                        <field name="product_id" options="{'no_quick_create': 1}" readonly="0" />
+                    </sheet>
+                </form>`,
+        });
```

Comment:
The relevance of this test (especially the scenario) is highly dependent on how the mock server operates, as it reproduces the issue if the mock server doesn't set the can_create attribute in this case (i.e. readonly="1" in the model + readonly="0" in the arch). This is worth a mock server test imho.